Magento_Catalog/templates/product/compare/list.phtml

```diff
@@ -156,23 +156,27 @@ $compareHelper = $this->helper(Magento\Catalog\Helper\Product\Compare::class);
                                     <?php endif; ?>
                                     <td class="compare__cell">
                                         <div class="attribute value">
-                                            <?php switch ($attribute->getAttributeCode()) {
-                                                case "price": ?>
-                                                    <?= $block->getProductPrice(
-                                                            $item,
-                                                            '-compare-list-' . $attribute->getCode()
-                                                        )
-                                                    ?>
-                                                <?php break;
-                                                case "small_image": ?>
-                                                    <?php $block->getImage($item, 'product_small_image')->toHtml(); ?>
-                                                <?php break;
-                                                default: ?>
-                                                    <?php if (is_string($block->getProductAttributeValue($item, $attribute))): ?>
-                                                        <?= $helper->productAttribute($item, $block->getProductAttributeValue($item, $attribute), $attribute->getAttributeCode()) ?>
-                                                    <?php endif; ?>
-                                                <?php break;
-                                            } ?>
+                                            <?php
+                                                $block->escapeHtml(
```

Comment:
This shouldn't be here.  Please check `magento/module-catalog/view/frontend/templates/product/compare/list.phtml` and add proper noEscape the way it is added in original file.